src/compiler/checker.ts

```diff
@@ -31361,6 +31361,27 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         );
     }
 
+    function discriminateContextualTypeByElements(node: ArrayLiteralExpression, contextualType: UnionType) {
+        const key = `D${getNodeId(node)},${getTypeId(contextualType)}`;
+        const cached = getCachedType(key);
+        if (cached) return cached;
+
+        // Create a discriminator for each element in the list
+        const discriminators = map(
+            node.elements,
+            (e, ndx) => [() => getContextFreeTypeOfExpression(e), "" + ndx as __String] as const,

```

Comment:
In the object version of this function there is a check for `isPossiblyDiscriminantValue`.  Is that necessary here?

Reply:
I'd expect both functions to be very similar so I'd say - yes. if `isPossiblyDiscriminantValue` is needed there - it's also needed here.